tests/infra/basicperf.py

```diff
@@ -137,28 +152,24 @@ def __call__(
                     )
 
 
-def configure_client_hosts(args, backups):
-    client_hosts = []
-    if args.one_client_per_backup:
-        assert backups, "--one-client-per-backup was set but no backup was found"
-        client_hosts = ["localhost"] * len(backups)
-    else:
-        if args.client_nodes:
-            client_hosts.extend(args.client_nodes)
-
-    if args.num_localhost_clients:
-        client_hosts.extend(["localhost"] * int(args.num_localhost_clients))
-
-    if not client_hosts:
-        client_hosts = ["localhost"]
-    return client_hosts
-
+def create_and_fill_key_space(size: int, primary: infra.node.Node) -> List[str]:
+    LOG.info(f"Creating and filling key space of size {size}")
+    space = [f"{i}" for i in range(size)]
+    with primary.client("user0") as c:
+        for key in space:
+            r = c.put(
+                f"/records/{key}",
+                body=f"{hashlib.md5(key.encode()).hexdigest()}",
+                headers={"content-type": "text/plain"},
+                log_capture=[],
+            )
+            assert r.status_code == http.HTTPStatus.NO_CONTENT.value, r
+    LOG.info("Key space created and filled")
+    return space
```

Comment:
Not critical, but this is going to be an unnecessary stall for larger key spaces - we should at some point add a single endpoint so this is a single, batched call.

Reply:
For the record, I did this with governance, but hit the limitation we place on governance not writing to private application tables (https://microsoft.github.io/CCF/main/audit/read_write_restrictions.html).

I'll move this to an application endpoint, in both cpp and js.

As discussed, I suspect we may re-open this discussion one day, specifically for situations where application key material is provisioned/re-keyed from governance. But we don't need to address this right now.